### ISSUE TITLE
Image Array

### ISSUE THREAD

**Issue by jeanieherold** [mladenilic/threesixty#4]
Hi @mladenilic

I really love how this works! It would be really nice to use an array of images rather than a sprite. Is that something that is doable with your ThreeSixty?

**Comment by jeanieherold** [comment]
I really appreciate you getting back to me! I do understand what you are
saying. And I did see in my explorations many people are using sprites.

What I have going is a work project. The images come from URLS and things
change up dynamically so I cannot use a sprite. There could many spinner
images and it needs to be able to happen dynamically. The backend and/or
company that generates these images will not but them in a sprite.

I converted Juan Cabrera's three60 that is also in github to work with an
array - he no longer supports it - just has it there for use with credit
given.

But your ThreeSixty is far superior in functionality, looks and user
experience - though at the time Juan made his it was probably as good as
anything out there (I do not know Juan and in no way mean anything
derogatory. His code really is good and it really does do a great job. I
was able to convert it to accepting an array very easily.

The speed of yours is perfect though. I am having trouble slowing down the
inertia on three60. I love how you just have the user put in a number for
speed with the ThreeSixty object you created.

I actually use the spinner in a FancyBox lightbox - load it in using iframe
set to preload false. It really does work great but now that I have seen
yours I was really hoping to use that if I could find a way to convert it
to accepting an array. It really is awesome.

If there is any chance you would like to explain to me the direction to go
in using an array rather than a sprite and seeing how it works loaded into
FancyBox through an iframe that handles the preload issue.... I would love
to try that. If it worked I would send you the code.

But I certainly understand if you have no interest in me trying that.

Either way... I very greatly appreciate you explaining and responding!

Thank you!
Jeanie

**Comment by mladenilic** [comment]
Hi Jeanie,

I get your problem. Okay, it makes sense to allow optionally passing array of images instead of sprite.

If you want, you can start working on it, once you open a PR, I can help you out. I'm currently busy with other projects, but I should be able to invest some time in this over the weekend.

As I said previously, it shouldn't require too much of a modification to make this work. You should take a look at `update` method. That's the one we need to change to load from image array.

**Comment by jeanieherold** [comment]
ok... cool. thanks!! I will work on it this week or sure. I will get back
to you!! thank you!!

**Comment by jeanieherold** [comment]
Hey :-)

I got it!! It works awesome.... except I have a flicker issue in the 1st
couple revolutions - probably while they are being cached in the browser- I
am not sure yet... but I cant swear its not the images I have...they are
really large... I am going to test some more on a different group of images
locally in the folder rather than absolute urls the rest of this week.

other than the flicker.. it is really smooth - at least on chrome and
safari. I still have quite a bit to test and maybe you might even know what
the flicker is or maybe that was even the issue you mentioned with arrays
vs sprites. But I think what I am seeing is fixable - just out of time
tonite to find it... I will keep looking and wait til the weekend to do a
pull request so I dont bug you anymore until I test more.

I did fork the repo - this is the link to what I have so far it you end up
having a look.
https://github.com/jeanieherold/threesixty/blob/master/src/threesixty.js

I did make sure and leave the image option. just created an array option. I
will test that wasnt messed up as well.

Have a great evening!
Jeanie

**Comment by jeanieherold** [comment]
Hey @mladenilic

I love how this works but it is unusable for anyone if I cant figure out
why it continually keeps requesting and returning a 304 rather a 200
success. Causes a horrible flicker with each update.

I think it is something in my code but I have NO idea what. It seems to do
well on IE and chrome and even mobile devices but is HORRIBLE on firefox
(and happens enough on chrome that i cant say it is just a browser issue).
I set up an example for you to see with home depot images that will be
pulled the same way I need them to at GEA where I work.

I couldnt upload our images to github - they are in development - so i
grabbed ones from the source code at HomeDepot website.

And the flicker with each GET request - sometimes just happens on the first
revolution but you hit refresh it gets 1000x worse. a hard refresh will
usually give the 200 response and the flicker stops.

I have set the image on an aqua background so that the flicker is very
evident.

I will send a pull request tonite.

Can you fix it?

Hoping...
Jeanie

**Comment by jeanieherold** [comment]
Hi @mladenilic

I have fixed the issue with the flickering - I refactored for img element rather than background image. example2.html for visual - It works rather nicely. If you would like to cancel that other pull requests I will send another PR. I worked off the same branch as the PR (sorry if I messed that up - I dont send PR often) - let me know if I need to refork the entire thing and start fresh with the updated code if you are interested in adding the feature to your program. 
https://github.com/jeanieherold/threesixty
